scripts/generate-releases.sh

```diff
@@ -2,64 +2,84 @@
+        echo "If provided commit SHA is less than 40 characters then it will be expanded to a full 40-characters long SHA. Default: currently checked out commit" >&2
+        echo "BRANCH - an optional parameter to specify git branch name for filtering snapshots by having branch name in annotations. Default: currently checked out branch" >&2
+        echo "" >&2
+        echo "You must have your KUBECONFIG point to the Konflux cluster, see https://spaces.redhat.com/pages/viewpage.action?pageId=407312060#HowtoeverythingKonfluxforRHACS-GettingocCLItoworkwithKonflux." >&2
+        return 1
+    fi
+}
+
+# Validate environment input and ensure it is either 'staging' or 'prod' (for master branch only).
+validate_environment() {
+    environment="$1"
+    branch="$2"
+    if [[ "${environment}" != "staging" && "${environment}" != "prod" ]]; then
+        echo "ERROR: ENVIRONMENT input must either be 'staging' or 'prod'." >&2
+        return 1
+    fi
+    if [[ "${environment}" == "prod" && "${branch}" != "master" ]]; then
+        echo "ERROR: prod release has to be done on master branch" >&2
+        return 1
+    fi
+}
+
+# Fetch the list of snapshots for provided commit and branch values. 
 # Make sure that only one the most recent snapshot per application is returned.
-snapshot_list="$(kubectl -n rh-acs-tenant get snapshot.appstudio.redhat.com -l pac.test.appstudio.openshift.io/sha="${COMMIT}" -o json | jq -r '
-  .items
-  | map(select((.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="'"${BRANCH}"'") or (.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="refs/heads/'${BRANCH}'")))
-  | sort_by(.spec.application)
-  | group_by(.spec.application)
-  | map(sort_by(.metadata.creationTimestamp) | last)
-  | .[]
-  | "\(.metadata.name)|\(.spec.application)"
-')"
-
-validate_input() {
+get_snapshots() {
+    commit="$1"
+    branch="$2"
+    kubectl get -n rh-acs-tenant snapshot -l pac.test.appstudio.openshift.io/sha="${commit}" -o json | jq -r '
+        .items
+        | map(select((.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="'"${branch}"'") or (.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="refs/heads/'${branch}'")))
+        | sort_by(.spec.application)
+        | group_by(.spec.application)
+        | map(sort_by(.metadata.creationTimestamp) | last)
+        | .[]
+        | "\(.metadata.name)|\(.spec.application)"'
+}
+
+# Validate all expected Snapshots were found.
+validate_snapshots() {
+    commit="$1"
+    snapshot_list="$2"
     pipelines_count="$(find ".tekton" -maxdepth 1 -type f -name "operator-index-ocp-*-build.yaml" | wc -l )"
     snapshots_count="$(echo "$snapshot_list" | wc -l )"
-    echo -e "found the following snapshots for \033[0;32m$COMMIT\033[0m commit in \033[0;32m$BRANCH\033[0m branch:" >&2
+
+    echo -e "found the following snapshots for \033[0;32m$commit\033[0m commit:" >&2
```

Comment:
Optional suggestion:
```suggestion
    echo -e "Found the following snapshots for \033[0;32m$commit\033[0m commit:" >&2
```